rasa/core/test.py

```diff
@@ -372,9 +372,20 @@ async def _create_data_generator(
     use_conversation_test_files: bool = False,
 ) -> "TrainingDataGenerator":
     from rasa.shared.core.generator import TrainingDataGenerator
+    from rasa.shared.constants import DEFAULT_DOMAIN_PATH
+    from rasa.model import get_model_subdirectories
+
+    core_model = None
+    if agent.model_directory:
+        core_model, _ = get_model_subdirectories(agent.model_directory)
+
+    if core_model and os.path.exists(os.path.join(core_model, DEFAULT_DOMAIN_PATH)):
+        domain_path = os.path.join(core_model, DEFAULT_DOMAIN_PATH)
+    else:
+        domain_path = None
 
     test_data_importer = TrainingDataImporter.load_from_dict(
-        training_data_paths=[resource_name]
+        training_data_paths=[resource_name], domain_path=domain_path
```

Comment:
Why can't we just use `DEFAULT_DOMAIN_PATH` like in NLU?

Reply:
Hmm, so this was something I wasn't sure about. In the Core case, the domain file is also stored inside the model file unlike with NLU, which is why NLU has to use the `DEFAULT_DOMAIN_PATH` but with Core it can be either one (model or default path). Does it make sense from a best practices perspective to keep it consistent here? I'm also thinking about the case when the domain file isn't in the default location, since the above will work regardless of where it's stored.

Reply:
hmm yeah it's a weird situation. I guess ideally we would always package the domain with the model. This seems fine for now though.